### ISSUE TITLE
Unable to easily check for equality between trait objects within an `Arc`

### ISSUE THREAD

**Issue by pwnorbitals** [rust-lang/rust#84239]
**I tried this code** ([reference StackOverflow question](https://stackoverflow.com/questions/67109860)) :

```rust
let a : Arc<dyn SomeTrait> = getA();
let b : Arc<dyn SomeTrait> = getB();

// The following two approaches don't work as expected
let eq1 = std::ptr::eq(a.as_ref(), b.as_ref()); 
let eq2 = Arc::ptr_eq(&a, &b);

```

**I expected to see this happen:** 
trait object comparison works as expected, with no false positives or false negatives. 

**Instead, this happened:**
 see StackOverflow explanation. Fat pointer comparison is a complex topic and I believe it's a papercut issue that the developper has to understand the internals of fat pointer layout to be able to check for equality

**Comment by jonas-schievink** [comment]
duplicate of https://github.com/rust-lang/rust/issues/63021

**Comment by pwnorbitals** [comment]
Then this would in turn be a duplicate of https://github.com/rust-lang/rust/issues/46139
